conda_lock/pypi_solver.py

```diff
@@ -214,8 +214,11 @@ def get_requirements(
 
             if op.package.source_type == "url":
                 url, fragment = urldefrag(op.package.source_url)
-                hash_type, hash = fragment.split("=")
-                hash = HashModel(**{hash_type: hash})
+                hash_splits = fragment.split("=")
+                if len(hash_splits) == 2:
+                    hash = HashModel(**{hash_splits[0]: hash_splits[1]})
+                else:
+                    hash = HashModel()
```

Comment:
Let's make sure all the cases are thoroughly covered:

```suggestion
                if fragment == "":
                    hash = HashModel()
                elif len(hash_splits) == 2:
                    hash = HashModel(**{hash_splits[0]: hash_splits[1]})
                else:
                    raise ValueError(f"Don't know what to do with {fragment}")
```